### PR TITLE
Fat content script architecture

### PR DESCRIPTION
**Pull request by muodov**
For: https://app.asana.com/0/72649045549333/1202153914495789/f
Tracking task: https://app.asana.com/0/0/1202457189191201/f

This PR moves most of the business logic into the content script, leaving only high-level orchestration to the background worker / native integration.

The changeset is rather big, consider reviewing commit-by-commit.

### CODE REVIEW

File: lib/cmps/sourcepoint-frame.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+  get isIntermediate(): boolean {
+    return false;
+  }
+
+  async detectCmp() {
+    if (window.top === window) {
+      // this rule is only for nested frames
+      return false;
+    }
```
Comment (sammacbeth) [inline lib/cmps/sourcepoint-frame.ts:26]: Can't you just use the `context` here?

File: lib/cmps/sourcepoint-top.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  get hasSelfTest(): boolean {
+    return true;
+  }
+
+  get isIntermediate(): boolean {
+    return true;
+  }
+
+  async detectCmp() {
+    return window.top === window && elementExists("div[id^='sp_message_container_']");
```
Comment (sammacbeth) [inline lib/cmps/sourcepoint-top.ts:20]: `context : { main: true, frame: false }`?

File: lib/cmps/trustarc-frame.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  get hasSelfTest(): boolean {
+    return false;
+  }
+
+  get isIntermediate(): boolean {
+    return false;
+  }
+
+  async detectCmp() {
+    return window.top !== window && location.href.startsWith("https://consent-pref.trustarc.com/?");
```
Comment (sammacbeth) [inline lib/cmps/trustarc-frame.ts:19]: Again - main frame only context should be used, right?

File: lib/cmps/trustarc-top.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  get isIntermediate(): boolean {
+    return !this._shortcutButton;
+  }
+
+  async detectCmp() {
+    if (window.top !== window) {
+      return false;
+    }
```
Comment (sammacbeth) [inline lib/cmps/trustarc-top.ts:37]: main frame context only.

**Review by sammacbeth — APPROVED** [review]

File: lib/cmps/sourcepoint-frame.ts (hunk 1 above)
Comment (muodov) [reply lib/cmps/sourcepoint-frame.ts]: Initially I implemented runContext only for declarative rules. But I've just pushed a change to do it for class-based rules too.

**Pull request merged**